### PR TITLE
Add fluentd chart

### PR DESCRIPTION
**Pull request by wilkers-steve**
This provides a functional deployment of fluentd. It currently
requires a service named elasticsearch-logging to function
properly. This will require additional configuration as logging
needs evolve, but functions in its current state

### CODE REVIEW

File: fluentd/templates/daemonset-fluentd.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      volumes:
+      - name: varlog
+        hostPath:
+          path: /var/log
+      - name: varlibdockercontainers
+        hostPath:
+          path: /var/lib/docker/containers
+      - name: etctdagent
+        emptyDir: {}
+      - name: tdagentconf
```
Comment (larryrensing) [inline fluentd/templates/daemonset-fluentd.yaml]: fluentd-etc

File: fluentd/templates/daemonset-fluentd.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        configMap:
+          name: fluentd-etc
+      - name: startsh
```
Comment (larryrensing) [inline fluentd/templates/daemonset-fluentd.yaml]: fluentd-bin

File: fluentd/templates/daemonset-fluentd.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          readOnly: true
+        - name: etctdagent
+          mountPath: /etc/td-agent
+        - name: tdagentconf
+          mountPath: /etc/td-agent/td-agent.conf
+          subPath: td-agent.conf
+        - name: startsh
+          mountPath: /tmp/start.sh
+          subPath: start.sh
+          readOnly: true
```
Comment (larryrensing) [inline fluentd/templates/daemonset-fluentd.yaml]: do we want volume overrides here?

File: fluentd/templates/daemonset-fluentd.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        hostPath:
+          path: /var/lib/docker/containers
+      - name: etctdagent
+        emptyDir: {}
+      - name: tdagentconf
+        configMap:
+          name: fluentd-etc
+      - name: startsh
+        configMap:
+          name: fluentd-bin
```
Comment (larryrensing) [inline fluentd/templates/daemonset-fluentd.yaml]: and here?

File: fluentd/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+description: A chart for fluentd aimed at use with elasticsearch
+name: fluentd
+version: 0.1.0
```
Comment (larryrensing) [inline fluentd/Chart.yaml]: add remainder of chart metadata (home, sources, maintainers, ..)

**Review by larryrensing — CHANGES_REQUESTED** [review]

File: fluentd/templates/daemonset-fluentd.yaml (hunk 3 above)
Comment (wilkers-steve) [reply fluentd/templates/daemonset-fluentd.yaml]: They're not appropriate here.  The only configuration file that needs to be mounted is td-agent.conf, and the only behavior we should support is the ability to override that file wholesale.

**Comment by alanmeadows** [comment]
Steve, should this be closed as it is now destined for github.com/openstack?

**Pull request closed without merging**